securesystemslib/keys.py

```diff
@@ -706,9 +711,9 @@ def create_signature(key_dict, data):
     private = binascii.unhexlify(private.encode('utf-8'))
     sig, scheme = ed25519_keys.create_signature(public, private, data, scheme)
 
-  # Continue to support keytypes of ecdsa-sha2-nistp256 and ecdsa-sha2-nistp384
+  # Continue to support keytypes of ecdsa-sha2-nistp256
   # for backwards compatibility with older securesystemslib releases
-  elif keytype in ['ecdsa', 'ecdsa-sha2-nistp256', 'ecdsa-sha2-nistp384']:
+  elif keytype in key_types_set:
```

Comment:
I'm not particularly happy about the unrelated changes in this module, at least not as part of this PR.
Even with your fixes the docstring is still inaccurate (e.g. *"cryptography library specified in 'settings'"*) and could be more concise (see my suggestions in the corresponding `SSlibSigner` docstring). And I'd also have a comment or two on the changes in the function body.

Would you mind just removing 67699608ea017c0a2c9c1dca51f03c2b589bbc53 from this PR? You can instead submit a new issue or add a comment e.g. to https://github.com/secure-systems-lab/securesystemslib/issues/271.

Reply:
Okay, I will remove the commit.
Added a comment about this: https://github.com/secure-systems-lab/securesystemslib/issues/271#issuecomment-782144697